pkg/cloudprovider/provider/kubevirt/provider.go

```diff
@@ -599,6 +599,10 @@ func (p *provider) newVirtualMachine(ctx context.Context, c *Config, pc *provide
 		}
 	}
 
+	// Add cluster labels
+	labels["cluster.x-k8s.io/cluster-name"] = c.Namespace
```

Comment:
IIRC, kubevirt ccm 0.58.0 adds the following selector to the infra svc:
```
selector:
  cluster.x-k8s.io/cluster-name: kubernetes
  cluster.x-k8s.io/role: worker
```

cluster-name is the cluster name, it's not `cluster-xyz` (that is the namespace name).
I think with this change it would not work.
Except if something was changed on KKP side to change the cluster name ? Sorry maybe I missed it last week, I was off.

Reply:
We first need to change in KKP 2 things before this PR is machine-controller can be merged:
- https://github.com/kubermatic/kubermatic/blob/main/pkg/resources/cloudcontroller.go#L109: add KubeVirt in the list of external CCM
- https://github.com/kubermatic/kubermatic/blob/main/pkg/resources/cloudcontroller/kubevirt.go#L134: the `--cluster-name` flag is set here. All other providers set `data.Cluster().Name`, do we really want to set a specific value for KubeVirt `data.Cluster(). NamespaceName`?